homeassistant/components/devolo_home_network/button.py

```diff
@@ -0,0 +1,137 @@
+    ),
+    PAIRING: DevoloButtonEntityDescription(
+        key=PAIRING,
+        icon="mdi:plus-network-outline",
+        name="Start PLC pairing",
+        press_func=lambda device: device.plcnet.async_pair_device(),  # type: ignore[union-attr]
+    ),
+    RESTART: DevoloButtonEntityDescription(
+        key=RESTART,
+        device_class=ButtonDeviceClass.RESTART,
+        entity_category=EntityCategory.CONFIG,
+        name="Restart device",
+        press_func=lambda device: device.device.async_restart(),  # type: ignore[union-attr]
+    ),
+    START_WPS: DevoloButtonEntityDescription(
+        key=START_WPS,
+        icon="mdi:wifi-plus",
+        name="Start WPS",
+        press_func=lambda device: device.device.async_start_wps(),  # type: ignore[union-attr]
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Get all devices and buttons and setup them via config entry."""
+    device: Device = hass.data[DOMAIN][entry.entry_id]["device"]
+
+    entities: list[DevoloButtonEntity] = []
+    if device.plcnet:
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[IDENTIFY],
+                device,
+            )
+        )
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[PAIRING],
+                device,
+            )
+        )
+    if device.device and "restart" in device.device.features:
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[RESTART],
+                device,
+            )
+        )
+    if device.device and "wifi1" in device.device.features:
+        entities.append(
+            DevoloButtonEntity(
+                entry,
+                BUTTON_TYPES[START_WPS],
+                device,
+            )
+        )
+    async_add_entities(entities)
+
+
+class DevoloButtonEntity(DevoloEntity, ButtonEntity):
+    """Representation of a devolo button."""
+
+    entity_description: DevoloButtonEntityDescription
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        description: DevoloButtonEntityDescription,
+        device: Device,
+    ) -> None:
+        """Initialize entity."""
+        self.entity_description = description
+        super().__init__(entry, device)
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        try:
+            await self.entity_description.press_func(self.device)
+        except DevicePasswordProtected as ex:
+            self.entry.async_start_reauth(self.hass)
+            raise HomeAssistantError(
+                f"Device {self.entry.title} is password protected"
+            ) from ex
+        except DeviceUnavailable as ex:
+            raise HomeAssistantError(
+                f"Device {self.entry.title} did not respond"
```

Comment:
Is config entry title really the name of the button?

Reply:
No, it's the name of the device that provides the functionality. So if I hit e.g. the Reboot button of the device "Test" and the device in not available the message will be "Device Test did not respond".

Reply:
I took a look and it seems to be the device hostname so that should be fine.